### PR TITLE
Trainers, event flags, mystery_event_msg, map object gfx

### PR DESCRIPTION
**Pull request by PikalaxALT**

### CODE REVIEW

File: src/renewable_hidden_items.c [hunk 1]
```diff
@@ -0,0 +1,609 @@
...
+        else
+            flags = sRenewableHiddenItems[i].common;
+        for (j = 0; j < 8; j++)
+        {
+            if (flags[j] != 0xFF)
+                FlagClear(FLAG_HIDDEN_ITEMS_START + flags[j]);
+        }
+    }
+}
+
```
Comment (jiangzhengwenjz) [inline src/renewable_hidden_items.c]: Redundant newline at the end

File: src/renewable_hidden_items.c [hunk 2]
```diff
@@ -0,0 +1,609 @@
...
+            if (common[j] != 0xFF)
+                FlagSet(FLAG_HIDDEN_ITEMS_START + common[j]);
+        }
+    }
+}
+
+void IncrementRenewableHiddenItemStepCounter(void)
+{
+    u16 var = VarGet(VAR_RENEWABLE_ITEM_STEP_COUNTER);
+    if (var < 1500) {
```
Comment (jiangzhengwenjz) [inline src/renewable_hidden_items.c]: bracket should start from newline

File: src/trainer_tower.c [hunk 3]
```diff
@@ -79,7 +71,6 @@ EWRAM_DATA struct UnkStruct_203F458 * gUnknown_203F458 = NULL;
 EWRAM_DATA struct UnkStruct_203F45C * gUnknown_203F45C = NULL;
 EWRAM_DATA u8 unused_variable = 0;
 
-void sub_815D96C(void);
 void sub_815DC8C(void);  // setup
 void sub_815DD2C(void);  // teardown
 void sub_815DD44(void);
```
Comment (jiangzhengwenjz) [inline src/trainer_tower.c]: A lot should be marked as static in this file, and stuff with external linkage should be declared in the header. And other things should be fixed like indent level for switch case (not essentially in this PR, tho)

Comment (PikalaxALT) [reply src/trainer_tower.c]: Will fix in a future PR

File: src/event_data.c [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+    gSpecialVar_0x8005 = 0;
+    gSpecialVar_0x8006 = 0;
+    gSpecialVar_0x8007 = 0;
+    gSpecialVar_0x8008 = 0;
+    gSpecialVar_0x8009 = 0;
+    gSpecialVar_0x800A = 0;
+    gSpecialVar_0x800B = 0;
+    gSpecialVar_Facing = 0;
+    gSpecialVar_Result = 0;
+    gSpecialVar_ItemId = 0;
```
Comment (jiangzhengwenjz) [inline src/event_data.c]: ITEM_NONE

File: include/quest_log.h [hunk 5]
```diff
@@ -15,7 +15,7 @@ struct UnkStruct_3005E90
 };
 
 extern u8 gUnknown_3005E88;
-extern u16 gUnknown_3005E8C;
+extern u16 sNumQuestLogs;
 extern struct UnkStruct_3005E90 gUnknown_3005E90;
 extern struct UnkStruct_203AE98 * gUnknown_3005E94;
```
Comment (jiangzhengwenjz) [inline include/quest_log.h]: These things are in COMMON but are they really used outside quest_log.c?

File: include/event_data.h [hunk 6]
```diff
@@ -60,7 +60,7 @@ extern u16 gSpecialVar_0x8014;
 extern u16 gSpecialVar_Result;
 extern u16 gSpecialVar_0x8004;
```
Comment (jiangzhengwenjz) [inline include/event_data.h]: redundant

File: data/event_scripts.s [hunk 7]
```diff
@@ -11,8 +11,14 @@
...
 #include "constants/spawn_points.h"
 #include "constants/field_effects.h"
+#include "constants/trainers.h"
+#include "constants/map_objects.h"
+#include "constants/fame_checker.h"
+#include "constants/seagallop.h"
 	.include "asm/macros.inc"
 	.include "asm/macros/event.inc"
+	.set FALSE, 0
+	.set TRUE,  1
```
[the remark this answers is not in the document]
Comment (PikalaxALT) [reply data/event_scripts.s]: Including constants/misc_constants.inc here causes assembler errors.

File: src/event_data.c (hunk 4 above)
Comment (PikalaxALT) [reply src/event_data.c]: That would be inconsistent with local context.

File: include/quest_log.h (hunk 5 above)
Comment (PikalaxALT) [reply include/quest_log.h]: shrug

File: src/data/trainer_graphics/back_pic_tables.h [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+const struct CompressedSpritePalette gTrainerBackPicPaletteTable[] = {
+    { gTrainerPalette_RedBackPic, 0 },
+    { gTrainerPalette_LeafBackPic, 1 },
+    { gTrainerPalette_RSBrendan1, 2 },
+    { gTrainerPalette_RSMay1, 3 },
+    { gTrainerPalette_PokeDudeBackPic, 4 },
+    { gTrainerPalette_OldManBackPic, 5 }
+};
+
```
Comment (jiangzhengwenjz) [inline src/data/trainer_graphics/back_pic_tables.h]: redundant newline

File: src/data/trainer_graphics/back_pic_anims.h [hunk 9]
```diff
@@ -0,0 +1,91 @@
+
```
Comment (jiangzhengwenjz) [inline src/data/trainer_graphics/back_pic_anims.h]: redundant newline

**Pull request merged**